### PR TITLE
Updated Docs Landing Page

### PR DESCRIPTION
**Pull request by phazonoverload**
## Scope

What's changed:

- Added sidebar to docs home for quick navigation to known pages.
- Cleaned up styling - spacing and light/dark toggling.
- Updated frameworks section to include new dev blog posts - will add the remaining 4 once complete. 

2024-01-17T1151 07
2024-01-17T1151 21

### CODE REVIEW

File: docs/.vitepress/components/Article.vue [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
+	alt?: string;
 }>();
 </script>
 
 <template>
 	<div class="article">
 		<a :href="url">
 			<div class="image">
-				<img :src="img" :alt="title" loading="lazy" />
+				<img :src="img" :alt="alt || ''" loading="lazy" />
```
Comment (phazonoverload) [inline docs/.vitepress/components/Article.vue:18]: To get ahead of this one - it is not useful to screenreaders to have the title as the alt and then the title immediately after.

File: docs/.vitepress/components/SnippetToggler.vue [hunk 2]
```diff
@@ -48,7 +47,7 @@ const changeSelected = async (choice: string, el: HTMLElement) => {
 </script>
 
 <template>
-	<div class="snippet-toggler" :class="{ dark: alwaysDark }">
```
Comment (paescuj) [inline docs/.vitepress/components/SnippetToggler.vue:51]: We should also clean-up the class itself

File: docs/.vitepress/components/Article.vue [hunk 3]
```diff
@@ -1,20 +1,21 @@
 <script setup lang="ts">
 defineProps<{
-	title: string;
+	title?: string;
```
Comment (paescuj) [inline docs/.vitepress/components/Article.vue:3]: Are there really cases when the title is undefined? If so we probably need to handle it, e.g. hiding the header

**Review by paescuj — CHANGES_REQUESTED** [review]
The layout needs a bit of padding clean-up, IMO.

### Before

Screenshot 2024-01-17 at 15-32-32 Directus Docs Directus Docs

### After

Experimental values for demonstration purposes:
- Horizontal padding set to 96px, removed the default padding for sidebar on the right side
- Top padding reduced to 64px

Screenshot 2024-01-17 at 15-40-15 Directus Docs Directus Docs

Ideally, we can align it with the blog index page as well as "normal" content pages.

File: docs/.vitepress/components/Article.vue (hunk 3 above)
Comment (phazonoverload) [reply docs/.vitepress/components/Article.vue:3]: I’m literally doing it on the homepage for the framework guides. But now I look at it, I suppose they could just be images

File: docs/home.css [hunk 4]
```diff
@@ -149,6 +142,30 @@
...
+}
+
+.frameworks .image:hover img {
+	scale: 1.05;
+}
+
+/* .frameworks img {
+	cursor: pointer;
+} */
+
```
Comment (paescuj) [inline docs/home.css:168]:
```suggestion
```

**Review by paescuj — APPROVED** [review]

**Pull request merged**